sunpy/visualization/animator/tests/test_basefuncanimator.py

```diff
@@ -95,6 +99,10 @@ def test_base_func_init(fig, colorbar, buttons):
     tfa._mouse_click(event)
     assert tfa.active_slider == 0
 
+    # Close the figure if it is a pyplot figure
+    if fig in [plt.figure(i) for i in plt.get_fignums()]:
+        plt.close(fig)
```

Comment:
Why not just do a `plt.close('all')` here?

Reply:
I think it's poor form to blindly close all figures because that could hide bugs or even create new problems (if there's some legitimate reason for a figure to be present on the stack).  If you're going to change the global state, it should be to intentionally restore it to the pre-test state: for example, you should use `plt.close('all')` only if the test explicitly starts with an assertion that the pyplot stack is initially empty (e.g., `assert not plt.get_fignums()`).